docs/1-getting-started/6-building-a-nav-bar/README.md

```diff
@@ -0,0 +1,161 @@
+# Building a Navigation Bar
+
+A navigation bar is a common component in web development. It allows users to navigate between different pages of a website. In this section, we will build a navigation bar for our blog site.
+
+## Page Tree
+
+Jahia supports creating pages in a tree structure. Each page can have child pages, forming a hierarchy. Create a few nested pages under the Home and About Us pages to create a tree structure. You can use the Page Composer sidebar to create and move pages around.
+
+![Nested pages](nested-pages.png)
+
+Our goal now is to build a navigation bar that displays the first two levels of the page tree. The navigation bar should be dynamic and update automatically when the page tree changes.
+
+We'll create a new component called `navBar` with a simple definition. Create the file `src/components/NavBar/defintion.cnd` with the following content:
+
+```cnd
+[hydrogen:navBar] > hydrogenmix:component, jnt:content
+```
+
+It doesn't have any properties, but you can add some if you want to customize the navigation bar further.
+
+Our `NavBar` component will create the following HTML structure:
+
+```html
+<nav>
+  <ul>
+    <li>
+      <a href="/home">Home</a>
+      <ul>
+        <li>
+          <a href="/home/new-products">New Products</a>
+        </li>
+        <li>
+          <a href="/home/case-studies">Case Studies</a>
+        </li>
+      </ul>
+    </li>
+    ...
+  </ul>
+</nav>
+```
+
+To get all these pages, we'll start at the site level, `renderContext.getSite()`, and iterate over all the child pages. We'll do it a second time to get the child pages of the child pages.
+
+Iterating over children can be done with the `getChildNodes` function. It takes a node and a filtering function as arguments. The filtering function should return `true` if the node should be included in the result. We'll only keep nodes of type `jnt:page`.
+
+Create the file `src/components/NavBar/NavBar.server.tsx` with the following content:
+
+```tsx
+import { buildNodeUrl, getChildNodes, jahiaComponent } from "@jahia/javascript-modules-library";
+import type { JCRNodeWrapper } from "org.jahia.services.content";
+
+/** Get all child pages of a node. */
+const getChildPages = (node: JCRNodeWrapper) =>
+  getChildNodes(node, -1, 0, (node) => node.isNodeType("jnt:page"));
+
+jahiaComponent(
+  {
+    componentType: "view",
+    nodeType: "hydrogen:navBar",
+    displayName: "NavBar",
+  },
+  (_, { renderContext, mainNode }) => (
+    <nav>
+      <ul>
+        {getChildPages(renderContext.getSite().getNode()).map((page) => (
```

Comment:
I can also update this utility to accept `JCRSiteNode`
```suggestion
        {getChildPages(renderContext.getSite().getNode(".")).map((page) => (
```

Reply:
Would love it!